homeassistant/components/bang_olufsen/config_flow.py

```diff
@@ -118,45 +127,22 @@ async def async_step_zeroconf(
         self._serial_number = discovery_info.properties[ATTR_SERIAL_NUMBER]
         self._beolink_jid = f"{discovery_info.properties[ATTR_TYPE_NUMBER]}.{discovery_info.properties[ATTR_ITEM_NUMBER]}.{self._serial_number}@products.bang-olufsen.com"
 
-        self.context["title_placeholders"] = {
-            "name": discovery_info.properties[ATTR_FRIENDLY_NAME]
-        }
+        return await self._create_entry()
 
+    async def _create_entry(self) -> FlowResult:
+        """Create the config entry for a discovered or manually configured Bang & Olufsen device."""
         await self.async_set_unique_id(self._serial_number)
         self._abort_if_unique_id_configured(updates={CONF_HOST: self._host})
```

Comment:
Move this to `async_step_zeroconf`, this should happen without user confirmation